### PR TITLE
feat(ayc): initial theme setup

### PR DESCRIPTION
**Pull request by lukeromanowicz**
- [x] I'm confirming that if I made any changes to public APIs or exposed any public APIs they are documented.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -136,5 +136,14 @@
     "@typescript-eslint/explicit-function-return-type": 0,
     "@typescript-eslint/no-unused-vars": 2,
     "@typescript-eslint/ban-ts-ignore": 0
-  }
+  },
+  "overrides": [
+    {
+      "files": "*.ts",
+      "rules": {
```
Comment (lukeromanowicz) [inline .eslintrc.json:143]: these rules are overlapped by @typescript-eslint/no-unused-vars and @typescript-eslint/no-undef

**Review by filrak — APPROVED** [review]

File: packages/about-you/api-client/src/index.ts [hunk 2]
```diff
@@ -24,14 +24,26 @@ let methods = {
...
+let locales = [];
 
 // TODO:: TEST THIS!!
 function override(overrides) {
   methods = { ...methods,
     ...overrides };
 }
 
-function setup() {
+function setup(setupConfig) {
```
Comment (krskibin) [inline packages/about-you/api-client/src/index.ts:40]: You create global variables with default values to override it with setupConfig props or themselves again.

File: packages/about-you/composables/src/composables/getters/cartGetters.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { CartGetters, AgnosticPrice, AgnosticTotals } from '@vue-storefront/interfaces';
+import { Cart, LineItem } from '@vue-storefront/boilerplate-api/src/types';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartItems = (cart: Cart): LineItem[] => [];
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/getters/cartGetters.ts:7]: there shouldn't be any problems with eslint when you're exporting these functions.

File: packages/about-you/composables/src/composables/getters/categoryGetters.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { CategoryGetters, AgnosticCategoryTree } from '@vue-storefront/interfaces';
+import { Category } from '@vue-storefront/boilerplate-api/src/types';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/getters/categoryGetters.ts:6]: same here

File: packages/about-you/composables/src/composables/getters/cartGetters.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartShippingPrice = (cart: Cart): number => 0;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartTotalItems = (cart: Cart): number => 0;
+
+const cartGetters: CartGetters<Cart, LineItem> = {
+  getTotals: getCartTotals,
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/getters/cartGetters.ts:47]: and you also use them here

File: packages/about-you/composables/src/composables/useCart/index.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+export const cart: Ref<BapiCart> = ref(null);
+
+// @todo: implement cart
+
+const params: UseCartFactoryParams<BapiCart, BapiCartItem, BapiProduct, BapiCoupon> = {
+  cart,
+  loadCart: async () => {
+    return {};
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/useCart/index.ts:17]: same here, but here this error should not occur anyway

File: packages/about-you/composables/src/composables/useUser/factoryParams.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+export const params: UseUserFactoryParams<BapiUser, any, any> = {
+  loadUser: async () => {
+    // @todo
+    return {};
+  },
+  logOut: async () => {
+    // @todo
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/useUser/factoryParams.ts:14]: same here

**Review by krskibin — CHANGES_REQUESTED** [review]

File: packages/about-you/composables/src/composables/getters/cartGetters.ts (hunk 3 above)
Comment (Qrzy) [reply packages/about-you/composables/src/composables/getters/cartGetters.ts:7]: There is - eslint will raise an error on unused function param ;)

Comment (krskibin) [reply packages/about-you/composables/src/composables/getters/cartGetters.ts:7]: that's right :D

**Review by krskibin — APPROVED** [review]

File: packages/about-you/api-client/src/index.ts (hunk 2 above)
Comment (krskibin) [reply packages/about-you/api-client/src/index.ts:40]: Ohhh, I did not notice that return is not inside the setup fn and also that local, country, etc are not global :exploding_head:

**Review by krskibin — APPROVED** [review]

**Review by patzick — APPROVED** [review]

**Pull request merged**